lib/ui/common/recommendation_compact_card/recommendation_compact_card.dart

```diff
@@ -0,0 +1,297 @@
+import 'package:farmsmart_flutter/ui/common/recommendation_card/recommendation_card_view_model.dart';
+import 'package:farmsmart_flutter/ui/common/roundedButton.dart';
+import 'package:farmsmart_flutter/ui/common/rounded_image_overlay.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+export 'package:farmsmart_flutter/ui/common/roundedButton.dart';
+
+class _Constants {
+  static String textDivider = '- ';
+  static EdgeInsets titlesPadding = const EdgeInsets.only(left: 21.0);
+  static EdgeInsets descriptionPadding = const EdgeInsets.only(top: 10);
+  static EdgeInsets actionsPadding = const EdgeInsets.only(top: 24);
+  static double horizontalActionsSeparation = 12;
+  static int titleFlex = 1;
+  static int subtitleFlex = 1;
+  static int titleMaxLines = 1;
+  static int subtitleMaxLines = 1;
+}
+
+class RecommendationCompactCardStyle {
+  final TextStyle titleTextStyle;
+  final TextStyle subtitleTextStyle;
+  final TextStyle descriptionTextStyle;
+  final RoundedButtonStyle leftActionButtonStyle;
+  final RoundedButtonStyle rightActionButtonStyle;
+  final double imageHeight;
+  final BorderRadiusGeometry imageBorderRadius;
+  final int descriptionMaxLines;
+  final EdgeInsets contentPadding;
+  final BoxDecoration rightActionBoxDecoration;
+  final Color overlayColor;
+  final double overlayIconHeight;
+  final double overlayIconWidth;
+  final String overlayIcon;
+
+  const RecommendationCompactCardStyle({
+    this.titleTextStyle,
+    this.subtitleTextStyle,
+    this.descriptionTextStyle,
+    this.leftActionButtonStyle,
+    this.rightActionButtonStyle,
+    this.imageHeight,
+    this.imageBorderRadius,
+    this.descriptionMaxLines,
+    this.contentPadding,
+    this.rightActionBoxDecoration,
+    this.overlayColor,
+    this.overlayIconHeight,
+    this.overlayIconWidth,
+    this.overlayIcon,
+  });
+
+  RecommendationCompactCardStyle copyWith({
+    TextStyle titleTextStyle,
+    TextStyle subtitleTextStyle,
+    TextStyle descriptionTextStyle,
+    RoundedButtonStyle leftActionButtonStyle,
+    RoundedButtonStyle rightActionButtonStyle,
+    double imageHeight,
+    BorderRadiusGeometry imageBorderRadius,
+    int descriptionMaxLines,
+    EdgeInsets contentPadding,
+    BoxDecoration rightBoxDecoration,
+    Color overlayColor,
+    double overlayIconHeight,
+    double overlayIconWidth,
+    String overlayIcon,
+  }) {
+    return RecommendationCompactCardStyle(
+      titleTextStyle: titleTextStyle ?? this.titleTextStyle,
+      subtitleTextStyle: subtitleTextStyle ?? this.subtitleTextStyle,
+      descriptionTextStyle: descriptionTextStyle ?? this.descriptionTextStyle,
+      leftActionButtonStyle:
+          leftActionButtonStyle ?? this.leftActionButtonStyle,
+      rightActionButtonStyle:
```

Comment:
I think the buttons need 2 styles. one normal and one disabled. The VM should then have a bool for if the button is disabled or not. (the same thing for the hero recommended card) The VM can't change styles. but the widget (view layer) can based on a VM var.

Reply:
@LeeMatthewHiggins DONE, you can check the RoundedButtonStateful, you can pass 2 styles for each button and it has a bool to display the first or the second (as talked)